pdns/dnsdist-tcp.cc

```diff
@@ -897,6 +897,10 @@ static void handleQuery(std::shared_ptr<IncomingTCPConnectionState>& state, stru
   gettime(&queryRealTime, true);
 
   auto query = reinterpret_cast<char*>(&state->d_buffer.at(0));
+  time_t* cachedValidTime = new time_t(-1);
+  char* queryBackup = new char[state->d_querySize];
```

Comment:
Can you please elaborate on the TCP issue?
What did you test, what was the problem, etc.

Reply:
I dont completely remember the problem but i remember it has something to do with the fact that TCP is managed in the OS level and you get a fd. This fd can get timedout and you can send message to unavailable fd. i dont remember the problem, since 2 months passed the last i worked on this, but i tested it with the original dnsdist program and after the cache timedout I tried to send a message so it will pass a query to the backend and it couldnt do it, same error.